### ISSUE TITLE
Consider making 'duplicate-id' rule as best practice

### ISSUE THREAD

**Issue by iamrafan** [dequelabs/axe-core#2808]
Currently ```duplicate-id``` rule is tagged as WCAG 4.1.1 and [F77](https://www.w3.org/TR/2016/NOTE-WCAG20-TECHS-20161007/F77) only mentions about how AT experience would be impacted if duplicate IDs are used.

We have tested the various ```duplicate-id``` failure scenarios for AT experience i,e. using screen readers like NVDA, JAWS, Narrator and Speech recognition software like Dragon NaturallySpeaking & Windows speech recognition. We found that none of the ```duplicate-id``` rule failures have an impact on these ATs and other failures which have an impact on ATs fall into either ```duplicate-id-aria``` or ```duplicate-id-active``` rule failure categories. 

Hence, please consider make the ```duplicate-id``` rule a best practice rule

Expectation: Tag ```duplicate-id``` rule as best practice

Actual: ```duplicate-id``` rule is tagged as WCAG 4.1.1

Motivation: There is no screen readers or speech recognition software experience impact on the elements that fail this rule

<pre><code>axe-core version: 4.1.1 </code></pre>

**Comment by WilcoFiers** [comment]
@iamrafan Duplicate IDs are not allowed by on any HTML element by WCAG 2 success criterion 4.1.1, which says:

> In content implemented using markup languages, elements have complete start and end tags, elements are nested according to their specifications, elements do not contain duplicate attributes, and **any IDs are unique**, except where the specifications allow these features.

This is a clear failure of WCAG. The argument for it is that it ensures a web page is robust, so that future assistive technologies won't be tripped up by it either. Whether or not we find that compelling isn't particularly relevant. This was agreed on in the standard, it was re-affirmed in WCAG 2.1, and unless something very strange happens it will be again in WCAG 2.2. Some work happened in WCAG 2.2 to take it out. I was involved in that. Unfortunately there wasn't enough momentum behind it, so it never made it into any of the drafts. 

Axe-core is a testing tool for WCAG, and since this is part of WCAG that can be automated, it should be in axe-core. We split the duplicate-id rule a while back to make it easy for people who aren't as concerned about WCAG normative to turn the rule off or change the tag on it. That's why there are three of them. You are very welcome to do that, but as long as this is part of the latest version of WCAG, I don't think we'll change this in axe-core.

**Comment by dylanb** [comment]
Last time I tested this with Dragon it was still a problem. Can you share test page(s), test procedures and versions of software please

**Comment by iamrafan** [comment]
@dylanb From below, please find the pages and versions of software we used for testing

### Test Pages: 
* [Flights from Seattle to Detroit - Bing](https://www.bing.com/search?q=flights+from+seattle+to+detroit&cvid=5c6dfefa28cd41ec905573e0d39d0da9&aqs=edge..69i57j0l6.410j0j1&pglt=515&FORM=ANNAB1&PC=U531)
* [India vs England cricket - Bing](https://www.bing.com/cricketdetails?q=India%20vs%20England%20cricket&IsCricketV3=1&QueryTimeZoneId=Pacific%20Standard%20Time&ResponseType=FullScore&Intent=None&Provider=SI&Lang=English&ScenarioName=SingleGame&TeamId0=4&GameId=199345&)
   * Navigate to commentary tab and Scan page
* [Cricket - Bing](https://www.bing.com/search?q=cricket)
   * Navigate to the International Cricket section and Scan the page
* [Microsoft Advertising Help Center](https://ads.microsoft.com/)
   * Create credentials, login and then scan the page



### ATs used for testing:
- Narrator - Windows 10, version 20H2
- JAWS Professional Edition - Version 2020.2006.12 ILM
- NVDA - Version 2020.4
- Dragon - Pro Group English_15.1
- Windows Speech Recognition - Windows 10, version 20H2

**Comment by WilcoFiers** [comment]
I'm going to close this issue. Deque's official position on this is that duplicate IDs anywhere in a page is a failure of WCAG 2.0, success criterion 4.1.1 Parsing. Axe-core is highly configurable, anyone who would prefer to run axe without this rule can turn it of.

**Comment by iamrafan** [comment]
@WilcoFiers sure, thanks for the update

**Comment by srai550** [comment]
https://www.w3.org/WAI/WCAG22/Understanding/parsing.html
Can be closed
